### ISSUE TITLE
Mouse emulation

### ISSUE THREAD

**Issue by Rigel88** [djlastnight/KeyboardSplitterXbox#73]
First of all, thanks for this great piece of software. It attempts to do what I've been doing by clumping 5 separate input mappers and emulators together. Sadly it falls just a little bit short in terms of mouse emulation. I've never owned a controller, so I can't really say if the way mouse movement is implemented here is good for games in general, but for a lot of games that have a free moving cursor (like top-down shooters) it would be much more convenient if the mouse was treated as such. 
The issues with current controls are:

- The mouse only has the same 8 degrees of freedom as a keyboard would. As far as I know, modern controllers allow for unlimited angles, as do mice.

- When the mouse is inactive, the cursor returns to central position. That's where mice and controllers differ the most. It's natural for controllers, but for mice it's game-breaking.

- Mouse sensitivity is way off, and there's no way to control it. I guess it depends on the hardware, but higher dpi mice are extremely jittery.

If there were, say, togglable options to address these issues, this program would be the only input mapper anyone would ever need in terms of Xinput. I know the program is called KEYBOARD Splitter, but there's nothing wrong with striving for perfection, amirite? Thanks for listening.

P.S. In case you need to have a look at the scripts I put together to emulate mouse movements: steamcommunity.com/sharedfiles/filedetails/?id=709640164

**Comment by djlastnight** [comment]
Hi. Thanks for your feedback. As you already noticed, the mouse support is limited. In case I have free time, I would try address the issues mentioned here.

**Comment by Bob36180** [comment]
Hi there !

I've seen that the program update added mouse support, but it does not work at all when I try to play with splitscreen, only keyboard works separately.
Is there a setting for this issue ? Is it normal ?
Thanks !

**Comment by djlastnight** [comment]
I never get into too much details about the mice stuff. This was experimental implementation, so do not expect anything mouse related to work properly/perfectly - there are many known issues.

**Comment by KeneePatel** [comment]
I would love to see this as well. I was looking for reWASD alternatives and that's where I found this repo. This project is amazing, kudos for that, but the only thing it lacks is this. Thank you and keep up the good work.

**Comment by Shankyorbis** [comment]
Are you planning to implement mouse functionality in the near future sir? I want to see how that will do when we can remap to the controller

**Comment by djlastnight** [comment]
Yep, at least I can try in case I feel boring during the holidays xD

**Comment by SubtlePineapple** [comment]
Did you make any progress on the mouse input? Super interested.
I can remap all required keys from keyboard and mouse to output as controller, but in game the mouse output is glitchy, depending on what i set the Axis id = 1, 2, 4, and 8 to (max or min %) it will either skip really far or not move at all, and I cant figure out a happy medium that feels smooth and consistent.

**Comment by GitAhmad304** [comment]
> Owner

how is the progress? in all honesty i downloaded this in the hopes of getting aim assist while playing on keyboard and mouse.
